app/models/weather_request.rb

```diff
@@ -3,8 +3,47 @@ class WeatherRequest < ApplicationRecord
 
   STATUSES = [PROCESSING = 'Processing',
               ERROR      = 'Error',
-              SUCCESS    = 'Success']
+              SUCCESS    = 'Success'].freeze
 
   validates :status, inclusion: { in: STATUSES }
-  
+
+  def weather_info
+    return nil if status != SUCCESS
+    hash_data = returned_json_as_hash
+    wind      = hash_data.dig('query', 'results', 'channel', 'wind')
+    astronomy = hash_data.dig('query', 'results', 'channel', 'astronomy')
+    condition = hash_data.dig('query', 'results', 'channel', 'item', 'condition')
+    OpenStruct.new(weather_time: condition['date'],
```

Comment:
Why did you use `OpenStruct`?

Reply:
I was very conflicted about whether to use models for the weather_info & weather_forecast data.  At first I could not think of a good reason to keep randomly collected weather data in models.  In the end I was leaning towards using models but I was running out of time.  I used `OpenStruct` so that the data would look the same to the users of WeatherRequest if I decided to switch them to models.  I guess I would normally return hashes but I thought that would make switching to models more difficult.  I'm still annoyed I didn't just make models out of the weather data.  If I had more time I would do that.

Reply:
That's all good.  I was also wondering if you had a specific reason for using `OpenStruct` over `Struct`.

Reply:
I forgot about Struct :)  I don't often use either, I was just thinking of making a temporary object and a quick google brought up OpenStruct first :)  Struct would have been more appropriate.